### PR TITLE
vedic 2.0.6 (new formula)

### PR DESCRIPTION
**Pull request by PtPrashantTripathi**
Co-author : @SumeetSinghJi 

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SumeetSinghJi** [comment]
The formula works but does not pass Brew guidelines which only allows building from source, not from binaries.

If you can help, inside your cargo.toml files the instructions you used to build it for MacOS arm64 and intel, I need to include those steps in the brew formula so that it builds from your source file directly on users PC, so you wont need individual binary.zip files.

the new formula will be lowercase vedic.rb and use the working template below

class Vedic < Formula
  desc "Sanskrit programming language written with Rust"
  homepage "https://github.com/vedic-lang/vedic"
  version "2.0.6"
  url "https://github.com/vedic-lang/vedic/archive/refs/tags/v#{version}.tar.gz"
  sha256 "5453386fcebfa48a5f3d39bc832d19edaaaa98d8d36841428ff6a20db0dd0151"
  
  livecheck do
    url :stable
    regex(/release[._-](\d+(?:\.\d+)+)/i)
    strategy :github_latest
  end

  def install
    # install instructions for vedic
  end

  def caveats
    <<~EOS
      To get started, run:
        vedic --help
    EOS
  end

  test do
    # Simple test to verify the binary is installed
    assert_match "Usage:", shell_output("#{bin}/vedic --help")
  end
end

**Comment by SumeetSinghJi** [comment]
Once I can get the install instructions on how you built those macos binaries (for both arm64 and intel) e.g., cargo toml dependencies etc., I will run the commands below to add them

____________________________________________________________________________________________________________

BREW WORKFLOW

READ MORE HERE: https://docs.brew.sh/Formula-Cookbook

# CLONE BREW LOCALLY 
# creates a clone of brew main repo isolated from local Brew on MacOS
brew tap --force homebrew/core
cd "$(brew --repository homebrew/core)"
cp ~/Documents/your_project/vedic.rb Formula/v
brew update
git remote add SumeetSinghJi https://github.com/SumeetSinghJi/homebrew-core.git
git checkout -b 2.0.6-vedic-formula origin/master

# BREW TEST PACKAGE WORKING
brew tests
HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source vedic
brew test vedic
brew audit --strict --online vedic

# BREW DELETE/CLEANUP PACKAGE
brew uninstall vedic
brew cleanup vedic
brew cleanup --prune=all
rm -rf $(brew --cache)/*
rm -rf $(brew --repository)/Library/Logs/Homebrew/*

# COMMIT + PUSH + PR (requires 2 new lines in commit message)
git add Formula/vedic.rb
git commit -m "vedic 2.0.6: add new formula


vedic lang is a Sanskrit Programming language.
It aims to assist native speakers, and learners to communiate
and program with Sanskrit characters with a fast and growing 
performance rich language and community to support them.
GitHub project located here: https://github.com/vedic-lang"
git push --set-upstream SumeetSinghJi 2.0.6-vedic-formula

**Comment by PtPrashantTripathi** [comment]
This is the script to build the mac binary

https://github.com/vedic-lang/vedic/blob/main/cicd%2Fbuild_mac.sh

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by PtPrashantTripathi** [comment]
> Once I can get the install instructions on how you built those macos binaries (for both arm64 and intel) e.g., cargo toml dependencies etc., I will run the commands below to add them. Once done I will create the documentation for you to keep forever and it will all be automated. The only thing you will ever need to do is update the formula shasum256 for the source file, and if the install instructions change for macos just update that. easy peasy.
> 
> BREW WORKFLOW
> 
> READ MORE HERE: https://docs.brew.sh/Formula-Cookbook
> 
> # CLONE BREW LOCALLY
> # creates a clone of brew main repo isolated from local Brew on MacOS
> brew tap --force homebrew/core cd "$(brew --repository homebrew/core)" cp ~/Documents/your_project/vedic.rb Formula/v brew update git remote add SumeetSinghJi https://github.com/SumeetSinghJi/homebrew-core.git git checkout -b 2.0.6-vedic-formula origin/master
> 
> # BREW TEST PACKAGE WORKING
> brew tests HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source vedic brew test vedic brew audit --strict --online vedic
> 
> # BREW DELETE/CLEANUP PACKAGE
> brew uninstall vedic brew cleanup vedic brew cleanup --prune=all rm -rf $(brew --cache)/* rm -rf $(brew --repository)/Library/Logs/Homebrew/*
> 
> # COMMIT + PUSH + PR (requires 2 new lines in commit message)
> git add Formula/vedic.rb git commit -m "vedic 2.0.6: add new formula
> 
> vedic lang is a Sanskrit Programming language. It aims to assist native speakers, and learners to communiate and program with Sanskrit characters with a fast and growing performance rich language and community to support them. GitHub project located here: https://github.com/vedic-lang" git push --set-upstream SumeetSinghJi 2.0.6-vedic-formula

https://github.com/vedic-lang/vedic/blob/main/cicd%2Fbuild_mac.sh

**Comment by PtPrashantTripathi** [comment]
@SumeetSinghJi or we can just setup a simple self hosted tap here
https://github.com/vedic-lang/homebrew-vedic

Whats your thought

**Comment by SumeetSinghJi** [comment]
Your the expert I just program as a hobby. I will fix, pass all tests and upload to you're new repo. Then you can decide. It will be a big achievement for you to have it available on brew worldwide, and it's self managing with little maintenance. Your choice. I'm proud of you regardless.

**Comment by SumeetSinghJi** [comment]
I cloned source code, ran chmod +x build_mac.sh
then ran with ./build_mac.sh
it created the release_mac folder with both zip files but corrupted. Error below. 

Could you advise if it's because my environment is missing something? Should I test by adding something to a copy of the build script as a test?

I'm trying to build the binaries from source but both me and brew formula are stuck at below. Any suggestions? Sorry I'm not a programmer.

_____________________________________________________________________________________


sumeetsingh@Sumeets-Air-2 cicd % ./build_mac.sh
   Compiling libc v0.2.140
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-apple-darwin` target may not be installed
  = help: consider downloading the target with `rustup target add x86_64-apple-darwin`

error: could not compile `libc` (lib) due to 1 previous error

Caused by:
  process didn't exit successfully: `rustc --crate-name libc --edition=2015 /Users/sumeetsingh/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libc-0.2.140/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=121 --crate-type lib --emit=dep-info,metadata,link -C opt-level=z -C linker-plugin-lto -C codegen-units=1 -C overflow-checks=on --cfg 'feature="default"' --cfg 'feature="std"' -C metadata=a31785666b74ea43 -C extra-filename=-a31785666b74ea43 --out-dir /Users/sumeetsingh/Documents/vedic/target/x86_64-apple-darwin/release/deps --target x86_64-apple-darwin -L dependency=/Users/sumeetsingh/Documents/vedic/target/x86_64-apple-darwin/release/deps -L dependency=/Users/sumeetsingh/Documents/vedic/target/release/deps --cap-lints allow --cfg freebsd11 --cfg libc_priv_mod_use --cfg libc_union --cfg libc_const_size_of --cfg libc_align --cfg libc_int128 --cfg libc_core_cvoid --cfg libc_packedN --cfg libc_cfg_target_vendor --cfg libc_non_exhaustive --cfg libc_long_array --cfg libc_ptr_addr_of --cfg libc_underscore_const_names --cfg libc_const_extern_fn` (signal: 5, SIGTRAP: trace/breakpoint trap)
tar: could not chdir to './target/x86_64-apple-darwin/release'

info: component 'rust-std' for target 'aarch64-apple-darwin' is up to date
    Finished `release` profile [optimized] target(s) in 0.01s
tar: could not chdir to './target/aarch64-apple-darwin/release'

sumeetsingh@Sumeets-Air-2 cicd %

**Comment by PtPrashantTripathi** [comment]
> I cloned source code, ran chmod +x build_mac.sh then ran with ./build_mac.sh it created the release_mac folder with both zip files but corrupted. Error below.
> 
> Could you advise if it's because my environment is missing something? Should I test by adding something to a copy of the build script as a test?
> 
> I'm trying to build the binaries from source but both me and brew formula are stuck at below. Any suggestions? Sorry I'm not a programmer.
> 
> sumeetsingh@Sumeets-Air-2 cicd % ./build_mac.sh Compiling libc v0.2.140 error[E0463]: can't find crate for `core` | = note: the `x86_64-apple-darwin` target may not be installed = help: consider downloading the target with `rustup target add x86_64-apple-darwin`
> 
> error: could not compile `libc` (lib) due to 1 previous error
> 
> Caused by: process didn't exit successfully: `rustc --crate-name libc --edition=2015 /Users/sumeetsingh/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libc-0.2.140/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=121 --crate-type lib --emit=dep-info,metadata,link -C opt-level=z -C linker-plugin-lto -C codegen-units=1 -C overflow-checks=on --cfg 'feature="default"' --cfg 'feature="std"' -C metadata=a31785666b74ea43 -C extra-filename=-a31785666b74ea43 --out-dir /Users/sumeetsingh/Documents/vedic/target/x86_64-apple-darwin/release/deps --target x86_64-apple-darwin -L dependency=/Users/sumeetsingh/Documents/vedic/target/x86_64-apple-darwin/release/deps -L dependency=/Users/sumeetsingh/Documents/vedic/target/release/deps --cap-lints allow --cfg freebsd11 --cfg libc_priv_mod_use --cfg libc_union --cfg libc_const_size_of --cfg libc_align --cfg libc_int128 --cfg libc_core_cvoid --cfg libc_packedN --cfg libc_cfg_target_vendor --cfg libc_non_exhaustive --cfg libc_long_array --cfg libc_ptr_addr_of --cfg libc_underscore_const_names --cfg libc_const_extern_fn` (signal: 5, SIGTRAP: trace/breakpoint trap) tar: could not chdir to './target/x86_64-apple-darwin/release'
> 
> info: component 'rust-std' for target 'aarch64-apple-darwin' is up to date Finished `release` profile [optimized] target(s) in 0.01s tar: could not chdir to './target/aarch64-apple-darwin/release'
> 
> sumeetsingh@Sumeets-Air-2 cicd %


If you have rust and Cargo already install then just run this command it will work
```sh
cargo build --package vedic --release
```

The bin file will be created inside 
"target" folder

File: Formula/v/vedic.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      To get started, run:
+        vedic --help
+    EOS
+  end
+
+  test do
+    # Simple test to verify the binary is installed
+    assert_match "Usage:", shell_output("#{bin}/vedic --help")
+  end
+end
```
Comment (chenrui333) [inline Formula/v/vedic.rb:59]:
```suggestion
class Vedic < Formula
  desc "vedic-lang is a Sanskrit programming language"
  homepage "https://vedic-lang.github.io/"
  url "https://github.com/vedic-lang/vedic/archive/refs/tags/v2.0.6.tar.gz"
  sha256 "5453386fcebfa48a5f3d39bc832d19edaaaa98d8d36841428ff6a20db0dd0151"
  license "MIT"
  head "https://github.com/vedic-lang/vedic.git", branch: "main"

  depends_on "rust" => :build

  def install
    system "cargo", "install", *std_cargo_args(path: "cli")
  end

  test do
    # hello world in vedic
    (testpath/"hello.ved").write <<~EOS
      वद("नमस्ते विश्व!");
    EOS

    assert_match "नमस्ते विश्व!", shell_output("#{bin}/vedic hello.ved")
  end
end

```

**Comment by chenrui333** [comment]
@PtPrashantTripathi you also need to turn on the push access so that the @BrewTestBot can publish the bottle into this PR.

**Comment by SumeetSinghJi** [comment]
I am 1 step ahead, it installs fine, however getting 2 brew errors that wont pass audit.

WORKING WITH 2 AUDIT ERRORS


sumeetsingh@Sumeets-Air-2 v % HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source vedic.rb
Error: Failed to load cask: vedic.rb
Cask 'vedic' is unreadable: wrong constant name #<Class:0x000000011e819b00>
Warning: Treating vedic.rb as a formula.

sumeetsingh@Sumeets-Air-2 v % brew audit --strict --online vedic                                      
vedic
  * line 21, col 5: use "cargo", "install", *std_cargo_args
Error: 1 problem in 1 formula detected.
sumeetsingh@Sumeets-Air-2 v % 

__________________________________________________________________________________________

REPLICATE ERROR STEPS:

brew tap --force homebrew/core
cd "$(brew --repository homebrew/core)"
cp ~/Documents/xxxxxxxxxx/vedic.rb Formula/v
cd Formula/v
HOMEBREW_VERBOSE=1 HOMEBREW_NO_AUTO_UPDATE=1 brew reinstall --build-from-source vedic.rb
brew test vedic
brew audit --strict --online vedic



__________________________________________________________________________________________

FORMULA: vedic.rb

class Vedic < Formula
  desc "Sanskrit programming language"
  homepage "https://vedic-lang.github.io/"
  url "https://github.com/vedic-lang/vedic/archive/refs/tags/v2.0.6.tar.gz"
  sha256 "5453386fcebfa48a5f3d39bc832d19edaaaa98d8d36841428ff6a20db0dd0151"
  license "MIT"
  head "https://github.com/vedic-lang/vedic.git", branch: "main"

  livecheck do
    url :stable
    regex(/release[._-](\d+(?:\.\d+)+)/i)
    strategy :github_latest
  end

  depends_on "rust" => :build

  def install
    arch = Hardware::CPU.arm? ? "aarch64-apple-darwin" : "x86_64-apple-darwin"

    # Build for the specific architecture
    system "cargo", "build", "--release", "--package", "vedic", "--target", arch

    # Install the built binary
    bin.install "target/#{arch}/release/vedic"
  end

  def caveats
    <<~EOS
      To get started, run:
        vedic --help
    EOS
  end

  test do
    # hello world in vedic
    (testpath/"hello.ved").write <<~EOS
      वद("नमस्ते विश्व!");
    EOS
    assert_match "नमस्ते विश्व!", shell_output("#{bin}/vedic hello.ved")
  end
end

**Comment by PtPrashantTripathi** [comment]
> @PtPrashantTripathi you also need to turn on the push access so that the @BrewTestBot can publish the bottle into this PR.

@chenrui333 what are steps for this?

**Comment by chenrui333** [comment]
> > @PtPrashantTripathi you also need to turn on the push access so that the @BrewTestBot can publish the bottle into this PR.
> 
> @chenrui333 what are steps for this?

follow this guide

please [allow maintainers to edit your PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to simplify the merge process.

**Comment by chenrui333** [comment]
there are some code style issues, can you run `brew style --fix vedic` and make another commit, after that, please all the commits and force push your branch.

**Comment by SumeetSinghJi** [comment]
Hi I tried this and it passes audit and test. If you look at the code you have to cd into cli folder first.

```
class Vedic < Formula
  desc "Simple Sanskrit programming language"
  homepage "https://vedic-lang.github.io/"
  url "https://github.com/vedic-lang/vedic/archive/refs/tags/v2.0.6.tar.gz"
  sha256 "5453386fcebfa48a5f3d39bc832d19edaaaa98d8d36841428ff6a20db0dd0151"
  license "MIT"
  head "https://github.com/vedic-lang/vedic.git", branch: "main"

  livecheck do
    url :stable
    regex(/release[._-](\d+(?:\.\d+)+)/i)
    strategy :github_latest
  end

  depends_on "rust" => :build

  def install
    cd "cli" do # use the cargo.toml file within the soure cli folder: ./vedic/cli/cargo.toml
      system "cargo", "install", *std_cargo_args
    end
  end

  def caveats
    <<~EOS
      Vedic-lang is a Sanskrit Programming Language

      Documentation here: https://vedic-lang.github.io/

      To get started, run:
        vedic --help
    EOS
  end

  test do
    # hello world in vedic
    (testpath/"hello.ved").write <<~EOS
      वद("नमस्ते विश्व!");
    EOS
    assert_match "नमस्ते विश्व!", shell_output("#{bin}/vedic hello.ved")
  end
end
```

File: Formula/v/vedic.rb
[the remark this answers is not in the document]
Comment (SumeetSinghJi) [reply Formula/v/vedic.rb:1]: If you look at my latest comment I have it working. I followed all guidelines e.g. can't use formula name as first word in description, and for installing you have to cd into the cli folder then run cargo install. Can you update the file? I've never done it before in a Pull Request not sure how.

[the remark this answers is not in the document]
Comment (PtPrashantTripathi) [reply Formula/v/vedic.rb:1]: Done but still failing with some Syntex issue. Leave it, I fix it, thanks for all the help.

[the remark this answers is not in the document]
Comment (SumeetSinghJi) [reply Formula/v/vedic.rb:1]: line 42 - you need a new line character at the end. if you are talking about this file: https://github.com/vedic-lang/homebrew-core/blob/master/Formula/v/vedic.rb

**Comment by chenrui333** [comment]
thanks for re-tagging, it would be nice to roll forward next time 👍

**Comment by SumeetSinghJi** [comment]
Okay fixed now

2 issues
1 - Path CLI instead of do cd
2 - Editing the Ruby text file in windows add's hidden carriage return "r" and "^M" text.

I ran: sumeetsingh@Sumeets-Air-2 v % sed -i '' 's/\r//' /Users/sumeetsingh/Documents/sandbox/vedic.rb

Future suggesstion: dont edit on Windows as it will add hidden text to macos/unix style text document
Also requires a new line at the end

I then ran all tests again with passes

```zsh
sumeetsingh@Sumeets-Air-2 v % brew test vedic
brew audit --strict --online vedic
brew audit --new vedic
brew style --fix vedic
==> Testing vedic
==> /opt/homebrew/Cellar/vedic/2.0.6/bin/vedic hello.ved

1 file inspected, no offenses detected
sumeetsingh@Sumeets-Air-2 v % 
```

```ruby
class Vedic < Formula
  desc "Simple Sanskrit programming language"
  homepage "https://vedic-lang.github.io/"
  url "https://github.com/vedic-lang/vedic/archive/refs/tags/v2.0.6.tar.gz"
  sha256 "5453386fcebfa48a5f3d39bc832d19edaaaa98d8d36841428ff6a20db0dd0151"
  license "MIT"
  head "https://github.com/vedic-lang/vedic.git", branch: "main"

  livecheck do
    url :stable
    regex(/release[._-](\d+(?:\.\d+)+)/i)
    strategy :github_latest
  end

  depends_on "rust" => :build

  def install
    system "cargo", "install", *std_cargo_args(path: "cli")
  end

  def caveats
    <<~EOS
      Vedic-lang is a Sanskrit Programming Language

      Documentation here: https://vedic-lang.github.io/

      To get started, run:
        vedic --help
    EOS
  end

  test do
    # hello world in vedic
    (testpath/"hello.ved").write <<~EOS
      वद("नमस्ते विश्व!");
    EOS
    assert_match "नमस्ते विश्व!", shell_output("#{bin}/vedic hello.ved")
  end
end

```

**Comment by SumeetSinghJi** [comment]
Sorry Prashant I'm just a engineer not a programmer, I created a new pull request in the Homebrew-vedic repo for the latest formula, It should be lowercase vedic.rb. Can you check first before you do the pull request?

**Pull request closed without merging**